DEVX-1474: Develop the demo docs into more of a playbook (#438)
confluentinc/examples: cloud-etl/docs/index.rst

==============

This demo showcases a cloud ETL solution leveraging all fully-managed services on `Confluent Cloud <https://confluent.cloud>`__.

> Using |ccloud| CLI, it creates a source connector that reads data from an AWS Kinesis stream into |ccloud|, then a |ccloud| KSQL application processes that data, and then a sink connector writes the output data into cloud storage in the provider of your choice (one of GCP GCS, AWS S3, or Azure Blob).

Reviewer: ```suggestion
Using |ccloud| CLI, the demo creates a source connector that reads data from an AWS Kinesis stream into |ccloud|, then a |ccloud| KSQL application processes that data, and then a sink connector writes the output data into cloud storage in the provider of your choice (one of GCP GCS, AWS S3, or Azure Blob).
```

"it" is a somewhat ambiguous subject

Warning
=======

> This demo uses real cloud resources, including that of |ccloud|, Amazon Kinesis, and one of the cloud storage providers.

Reviewer: Line 10 above is AWS Kinesis. Should that be Amazon Kinesis as well?

Author: @tlblessing good suggestion, thank you

Setup
-----

> Because this demo interacts with real resources in Kinesis, a destination storage service, and |ccloud|, you must setup some initial parameters to communicate with these services.

Reviewer: ```suggestion
Because this demo interacts with real resources in Kinesis, a destination storage service, and |ccloud|, you must set up some initial parameters to communicate with these services.
```

      ksql.endpoint=https://<KSQL ENDPOINT>
      ksql.basic.auth.user.info=<KSQL API KEY>:<KSQL API SECRET>

>    To retrieve the values for the endpoints and credentials in the file above, find them either via the |ccloud| UI or |ccloud| CLI commands. If you have multiple |ccloud| clusters, make sure to use the one with the associated KSQL cluster.  The commands below demonstrate how to retrieve the values using the |ccloud| CLI.

Reviewer: ```suggestion
   To retrieve the values for the endpoints and credentials in the file above, find them using either the |ccloud| UI or |ccloud| CLI commands. If you have multiple |ccloud| clusters, make sure to use the one with the associated KSQL cluster.  The commands below demonstrate how to retrieve the values using the |ccloud| CLI.
```

Avoid via per style guides

     cd cloud-etl

#. Modify the demo configuration file at :devx-examples:`config/demo.cfg|cloud-etl/config/demo.cfg`. Set the proper credentials and parameters for the source, e.g. AWS Kinesis.  Also set the required parameters for the respective destination cloud storage provider:

   - GCP GCS

     - ``DESTINATION_STORAGE='gcs'``
     - ``GCS_CREDENTIALS_FILE``
     - ``GCS_BUCKET``

   - AWS S3

     - ``DESTINATION_STORAGE='s3'``
     - ``S3_PROFILE``
     - ``S3_BUCKET``

   - Azure Blob

     - ``DESTINATION_STORAGE='az'``
     - ``AZBLOB_STORAGE_ACCOUNT``
     - ``AZBLOB_CONTAINER``

Run
---

> #. Log into |ccloud| with the command ``ccloud login``, and use your |ccloud| username and password.

Reviewer: ```suggestion
#. Log in to |ccloud| with the command ``ccloud login``, and use your |ccloud| username and password.
```

   .. code:: shell

      ccloud login --url https://confluent.cloud

#. Run the demo. It takes approximately 7 minutes to run.

   .. code:: bash

      ./start.sh

Validate
--------

#. Using the `Confluent Cloud CLI <https://docs.confluent.io/current/quickstart/cloud-quickstart/index.html#step-2-install-the-ccloud-cli>`__, list all the fully-managed connectors created in this cluster.

   .. code:: bash

      ccloud connector list      

   Sample output:

   .. code:: bash

           ID     |         Name         | Status  |  Type   
      +-----------+----------------------+---------+--------+
        lcc-knjgv | demo-KinesisSource   | RUNNING | source  
        lcc-nwkxv | demo-GcsSink-avro    | RUNNING | sink    
        lcc-3r7w2 | demo-GcsSink-no-avro | RUNNING | sink    

> #. These connectors are created automatically by the demo using the |ccloud| CLI command ``ccloud connector create`` and passing in the appropriate configuration file from the :devx-examples:`connectors directory|cloud-etl/connectors/`.  Describe one of the connectors in more detail.

Reviewer: ```suggestion
#. The demo automatically creates these connectors using the |ccloud| CLI command ``ccloud connector create`` that passes in the appropriate configuration file from the :devx-examples:`connectors directory|cloud-etl/connectors/`.  Describe one of the connectors in more detail.
```

Suggestion for more parallel and active verb tense.

   .. figure:: images/flow.png
      :alt: image

#. The demo's :devx-examples:`KSQL commands|cloud-etl/ksql.commands` generated a KSQL TABLE ``COUNT_PER_SOURCE``, formatted as JSON, and its underlying Kafka topic is ``COUNT_PER_SOURCE``. It also generated a KSQL TABLE ``SUM_PER_SOURCE``, formatted as Avro, and its underlying Kafka topic is ``SUM_PER_SOURCE``. View its schema in |sr-ccloud|.

   .. code:: bash

      curl --silent -u <SR API KEY>:<SR API SECRET> https://<SR ENDPOINT>/subjects/SUM_PER_SOURCE-value/versions/latest | jq -r '.schema' | jq .

   Sample output:

   .. code:: json

      {
        "type": "record",
        "name": "KsqlDataSourceSchema",
        "namespace": "io.confluent.ksql.avro_schemas",
        "fields": [
          {
            "name": "EVENTSOURCEIP",
            "type": [
              "null",
              "string"
            ],
            "default": null
          },
          {
            "name": "KSQL_COL_1",
            "type": [
              "null",
              "long"
            ],
            "default": null
          }
        ]
      }

> #. View the data from Kinesis, Kafka, and cloud storage after running the demo.  Sample output shown below:

Reviewer: ```suggestion
#. View the data from Kinesis, |ak|, and cloud storage after running the demo.  Sample output shown below:
```
